packages/plugin-core/src/commands/CreateSchemaFromHierarchyCommand.ts

```diff
@@ -132,7 +131,7 @@ function createCandidatesMapByFname(items: readonly SchemaCandidate[]) {
 function getUriFromSchema(schema: SchemaModuleProps) {
   const vaultPath = vault2Path({
     vault: schema.vault,
-    wsRoot: getDWorkspace().wsRoot,
+    wsRoot: ExtensionProvider.getDWorkspace().wsRoot,
```

Comment:
While waiting for the command injection refactor change, should we go ahead and create a special constructor for `CreateSchemaFromHierarchyCommand` with ISchemaSyncService/IDendronExtension injected? 

If you think it's better to wait until later, that's also fine.

Reply:
Since using ExtensionProvider is so much less invasive and since we need to touch ALOT of code I am kinda leaning towards the approach of doing a pass through the code base of using ExtensionProvider wherever it's convenient (for the sake speedy untangle), and once we have broken all the circles we can pass IDendronExtension from constructor.